### PR TITLE
XIVY-12783 switch to axios

### PR DESCRIPTION
**Pull request by ivy-lmu**

### CODE REVIEW

File: src/engine/engine-manager.ts [hunk 1]
```diff
@@ -74,18 +74,14 @@ export class IvyEngineManager {
     });
   }
 
-  public async initProjects() {
-    if (await this.devContextPath) {
-      const ivyProjectDirectories = await this.ivyProjectDirectories();
-      await this.ivyEngineApi.initProjects(ivyProjectDirectories);
-    }
+  private async initProjects(): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:77]: no need to define return statement. Should be infered from the content of this function

File: src/engine/engine-manager.ts [hunk 2]
```diff
@@ -74,18 +74,14 @@ export class IvyEngineManager {
...
+    const ivyProjectDirectories = await this.ivyProjectDirectories();
+    await this.ivyEngineApi.initProjects(ivyProjectDirectories);
   }
 
-  public async deployProjects() {
-    if (await this.devContextPath) {
-      const ivyProjectDirectories = await this.ivyProjectDirectories();
-      await this.ivyEngineApi.deployProjects(ivyProjectDirectories);
-    }
+  public async deployProjects(): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:82]: no need to define return statement.

File: src/engine/engine-manager.ts [hunk 3]
```diff
@@ -96,46 +92,36 @@ export class IvyEngineManager {
     await this.mavenBuilder.buildProject(ivyProjectDirectory);
   }
 
-  public async deployProject(ivyProjectDirectory: string) {
-    if (await this.devContextPath) {
-      await this.ivyEngineApi.deployProjects([ivyProjectDirectory]);
-    }
+  public async deployProject(ivyProjectDirectory: string): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:95]: no need to define return statement.

File: src/engine/engine-manager.ts [hunk 4]
```diff
@@ -96,46 +92,36 @@ export class IvyEngineManager {
...
+    await this.ivyEngineApi.deployProjects([ivyProjectDirectory]);
   }
 
-  public async buildAndDeployProjects() {
-    if (await this.devContextPath) {
-      const ivyProjectDirectories = await this.ivyProjectDirectories();
-      await this.buildProjects();
-      await this.ivyEngineApi.deployProjects(ivyProjectDirectories);
-    }
+  public async buildAndDeployProjects(): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:99]: no need to define return statement.

File: src/engine/engine-manager.ts [hunk 5]
```diff
@@ -96,46 +92,36 @@ export class IvyEngineManager {
...
+    await this.buildProjects();
+    await this.ivyEngineApi.deployProjects(ivyProjectDirectories);
   }
 
-  public async buildAndDeployProject(ivyProjectDirectory: string) {
-    if (await this.devContextPath) {
-      await this.buildProject(ivyProjectDirectory);
-      await this.ivyEngineApi.deployProjects([ivyProjectDirectory]);
-    }
+  public async buildAndDeployProject(ivyProjectDirectory: string): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:105]: no need to define return statement.

File: src/engine/engine-manager.ts [hunk 6]
```diff
@@ -96,46 +92,36 @@ export class IvyEngineManager {
...
+    await this.buildProject(ivyProjectDirectory);
+    await this.ivyEngineApi.deployProjects([ivyProjectDirectory]);
   }
 
-  public async createProcess(newProcessParams: NewProcessParams) {
-    if (await this.devContextPath) {
-      await this.createAndOpenProcess(newProcessParams);
-      await this.ivyEngineApi.deployProjects([newProcessParams.path]);
-    }
+  public async createProcess(newProcessParams: NewProcessParams): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:110]: no need to define return statement.

File: src/engine/engine-manager.ts [hunk 7]
```diff
@@ -144,10 +130,8 @@ export class IvyEngineManager {
     executeCommand('vscode.open', vscode.Uri.parse(newProcessUri));
   }
 
-  public async deleteProject(ivyProjectDirectory: string) {
-    if (await this.devContextPath) {
-      this.ivyEngineApi.deleteProject(ivyProjectDirectory);
-    }
+  public async deleteProject(ivyProjectDirectory: string): Promise<void> {
```
Comment (ivy-lli) [inline src/engine/engine-manager.ts:133]: no need to define return statement.

**Review by ivy-lli — CHANGES_REQUESTED** [review]

File: src/engine/api/engine-api.ts [hunk 8]
```diff
@@ -10,37 +12,27 @@ import {
...
+} from './project-request';
 
 export class IvyEngineApi {
   private readonly API_PATH = 'api/web-ide';
-  private engineUrl: URL;
-  private devContextPath = '';
-
-  constructor(engineUrlString: string) {
-    this.engineUrl = new URL(engineUrlString);
+  private readonly _devContextPath: Promise<string>;
```
[the remark this answers is not in the document]
Comment (ivy-lmu) [reply src/engine/api/engine-api.ts:21]: because there is a getter `get devContextPath` that's the default auto generation pattern for getters

**Review by ivy-lli — APPROVED** [review]
Looks good. Thanks for the improvements

**Pull request merged**